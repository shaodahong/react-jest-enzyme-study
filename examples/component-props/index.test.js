import React from 'react';
import ComponentProps from './index';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, render, mount } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

it('ComponentProps test', () => {
  const wrapper = shallow(<ComponentProps />);
  expect(wrapper).toMatchSnapshot();

  const isShow = true;

  wrapper.setProps({
    isShow
  });

  expect(wrapper.hasClass('show')).toBe(isShow);
});
