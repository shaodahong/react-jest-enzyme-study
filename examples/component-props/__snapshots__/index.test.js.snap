// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentProps test 1`] = `
<div
  className="hide"
>
  hide
</div>
`;
